orchestrator-salt/src/main/resources/salt/salt/postgresql/disaster_recovery/scripts/backup_db.sh

```diff
@@ -27,26 +27,71 @@ doLog() {
   echo "$(date "+%Y-%m-%dT%H:%M:%SZ") $type_of_msg ""$msg" >>$LOGFILE
 }
 
-if [[ $# -lt 6 || $# -gt 10 || "$1" == "None" ]]; then
-  doLog "Invalid inputs provided"
-  doLog "Script accepts at least 6 and at most 7 inputs:"
-  doLog "  1. Object Storage Service url to place backups."
-  doLog "  2. PostgreSQL host name."
-  doLog "  3. PostgreSQL port."
-  doLog "  4. PostgreSQL user name."
-  doLog "  5. Ranger admin group."
-  doLog "  6. Whether or not to close connections for the database while it is being backed up."
-  doLog "  7-10. (optional) Names of the databases to backup. If not given, will backup ranger and hive databases."
-  exit 1
-fi
+log_invalid_input_reason() {
+  doLog "This might be caused by the command is not running on the master Getaway node or there has never had a backup/restore from the CDP CLI run."
+}
 
-BACKUP_LOCATION="$1"
-HOST="$2"
-PORT="$3"
-USERNAME="$4"
-RANGERGROUP="$5"
-CLOSECONNECTIONS="$6"
-DATABASENAMES="${@: 7}"
+# Print out all required inputs if they are passed in.
+while getopts ":s:h:p:u:g:c:n:" OPTION; do
```

Comment:
I understand why you're changing this to not use positional arguments, but please note that this will break backwards compatibility in case any customers have their own scripts for running backup/restore. Please confirm with Kalyan that this is okay.

Please also make sure that running a backup/restore via the CB flow is still functional. You can check this by running a resize and making sure the backup/restore steps are successful.

Reply:
@sxxgrc That's a really good point - I didn't realize customers will run with their own scripts. Let me ask Kalyan about it. Thank you! I will update this thread after getting confirmation and testing with resize. If we do need to think of customers who use their own scripts, I will switch back to the first version PR which only has a few changes. https://github.com/hortonworks/cloudbreak/pull/14308/files